src/coreclr/tools/Common/Internal/Text/Utf8StringBuilder.cs

```diff
@@ -16,10 +16,10 @@ public Utf8StringBuilder()
         {
         }
 
-        // TODO: This should return ReadOnlySpan<byte> instead once available
-        public byte[] UnderlyingArray => _buffer;
         public int Length => _length;
 
+        public ReadOnlySpan<byte> AsSpan() => _buffer.AsSpan(0, _length);
```

Comment:
Should we perhaps mirror ValueStringBuilder here and add ``terminate``?
https://github.com/dotnet/runtime/blob/34e65b9e6c3100bbbebcd78fd03f3b2e88cdc9f0/src/libraries/Common/src/System/Text/ValueStringBuilder.cs#L104-L112

Reply:
Are there any places where this would be useful?

Reply:
Only one place, the Dwarf CIE frame serialization 😅 Probably not worth it on second thought.

Reply:
It would not work well there. 

The `terminate` argument is specifically designed to create null-terminated strings for interop with native code. It appends the null-terminator, but it does not include the null terminator in the length (same deal as what you get from `u8` string literals in C#). Dwarf CIE frame serialization would need the null terminator included in the length.